plugins/woocommerce-blocks/tests/e2e/tests/templates/template-customization.block_theme.spec.ts

```diff
@@ -149,32 +155,37 @@ test.describe( 'Template customization', () => {
 				// duplicate templates with the same name.
 				// See: https://github.com/woocommerce/woocommerce/issues/42220
 				await admin.visitSiteEditor( {
-					path: `/${ testData.templateType }/all`,
+					postType: testData.templateType,
 				} );
 
 				await page
 					.getByPlaceholder( 'Search' )
 					.fill( testData.templateName );
 
-				const templateRow = page.getByRole( 'row' ).filter( {
-					has: page.getByRole( 'link', {
-						name: testData.templateName,
-						exact: true,
-					} ),
-				} );
-				const resetButton = templateRow.getByLabel( 'Reset', {
-					exact: true,
-				} );
-				const revertedNotice = page
+				const resetNotice = page
 					.getByLabel( 'Dismiss this notice' )
-					.getByText( `"${ testData.templateName }" reverted.` );
+					.getByText(
+						testData.templateType === 'wp_template'
+							? `"${ testData.templateName }" reset.`
+							: `"${ testData.templateName }" deleted.`
+					);
 				const savedButton = page.getByRole( 'button', {
 					name: 'Saved',
 				} );
 
-				await resetButton.click();
+				// Wait until search has finished.
+				await page.waitForFunction( ( templatesLength ) => {
+					return (
+						document.querySelectorAll( '[aria-label="Actions"]' )
+							.length < templatesLength
+					);
+				}, CUSTOMIZABLE_WC_TEMPLATES.length );
```

Comment:
This can be replaced with an assertion:

```suggestion
				// Wait until search has finished.
				await expect( page.getByLabel( 'Actions' ) ).toHaveLength(
					CUSTOMIZABLE_WC_TEMPLATES.length
				);
```

Reply:
What this function does is to check that the list of templates displayed on screen is "less than" the total number of WooCommerce templates. That was a simple way I found to wait until the search was performed correctly.

In the code from your suggestion, my understanding is that we would assert that the number of templates listed matches the number of WooCommerce templates, which is not exactly the same, no? :thinking:

Reply:
Ah, good catch – I saw it as `wait until less than`! In such case, I suggest providing an exact number of the expected `Actions` count when the search is done, e.g.:

```suggestion
				// Wait until search has finished.
				await expect( page.getByLabel( 'Actions' ) ).toHaveCount( 4 );
```

It should always be the same number, and this way, we're sure it's always correct and not, for example, and intermediate debounced value. Does that make sense?

Reply:
The issue we have is that in most cases there is going to be only one result. Ie, if you search for "Product Search Results" it will only display one template. There is one exception where we get two results: searching for "Single Product" returns the "Single Product" template and the customized "Single Product: Belt" one. I could have hardcoded `1` and `2` depending on the test, but I thought doing a generic "wait until search has finished" was more future-proof, even though I can see how it might be more confusing.

How would you approach this case where in most tests it's `1` except for the Single Product custom template where it's `2`?

Reply:
Ah, I see. If the stability of the search is not an issue (no intermediate values), then I think we can go with `expect.poll` on this one:

```suggestion
				// Wait until search has finished.
				const searchResults = page.getByLabel( 'Actions' );
				await expect
					.poll( async () => await searchResults.count() )
					.toBeLessThan( CUSTOMIZABLE_WC_TEMPLATES.length );
```

Reply:
Awesome, it's definitely much cleaner. :100: